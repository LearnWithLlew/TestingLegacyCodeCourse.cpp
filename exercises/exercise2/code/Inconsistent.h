#pragma once
#include <cstdlib>
#include <ctime>
#include <string>
#include <iostream>
#include <sstream>
#include <ctime>
#include <chrono>

class Inconsistent
{
public:
    int getID()
    {
        srand(time(0));
        return (rand() % 10 + 1);
    }

    std::string print()
    {
        std::stringstream stream;
        auto end = std::chrono::system_clock::now();
        std::time_t end_time =
                std::chrono::system_clock::to_time_t(end);

        stream << "started at " << std::ctime(&end_time);
        return stream.str();
    }
};
