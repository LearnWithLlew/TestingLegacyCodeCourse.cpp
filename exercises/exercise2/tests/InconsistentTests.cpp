#include <catch2/catch.hpp>
#include "ApprovalTests.hpp"

#include "Inconsistent.h"

using namespace ApprovalTests;

TEST_CASE("Poke GetID")
{
	Inconsistent subject;
	
	Approvals::verify(subject.getID(0));
}

TEST_CASE("Poke print")
{
}

TEST_CASE("Scrub print")
{
}
