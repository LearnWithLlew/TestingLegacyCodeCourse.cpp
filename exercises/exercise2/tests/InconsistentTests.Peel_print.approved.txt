started at Tue Feb  1 03:04:05 2000 by name
